test/regex.jl

```diff
@@ -70,6 +70,7 @@
     @test count(r"\w+", "foo bar", overlap=true) == 6
     @test count(r"\w*", "foo bar") == 4
     @test count(r"\b", "foo bar") == 4
+    @test count('a',"batman") == 2
```

Comment:
Looks good. I think it would be nice to have a couple more tests. E.g. empty string or `overlap = true`

Reply:
I added those now. Overlap is not possible with Char, as they always have length 1. 

One thing I am thinking is that the `findall` function here suffers from the same problem.

Reply:
Sure, overlap is silly here. However it is a corner case, that does not hurt testing I think. Anyway thanks for adding these!

Reply:
I would at least also search for a non-ASCII character that is two or more code units wide.
```suggestion
    @test count('a', "batman") == 2
```
Also: use spaces consistent with surrounding code.